src/main/java/org/trebol/integration/payments/webpayplus/WebpayplusPaymentServiceImpl.java

```diff
@@ -1,112 +1,112 @@
+import cl.transbank.webpay.exception.TransactionCreateException;
+import cl.transbank.webpay.webpayplus.WebpayPlus;
+import cl.transbank.webpay.webpayplus.responses.WebpayPlusTransactionCreateResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.trebol.integration.IPaymentsIntegrationService;
+import org.trebol.integration.exceptions.PaymentServiceException;
+import org.trebol.pojo.PaymentRedirectionDetailsPojo;
+import org.trebol.pojo.SellPojo;
+
+import java.io.IOException;
+
+@Service
+public class WebpayplusPaymentServiceImpl
+  implements IPaymentsIntegrationService {
+
+  private final Logger logger = LoggerFactory.getLogger(WebpayplusPaymentServiceImpl.class);
+  private final WebpayplusPaymentProperties properties;
+
+  @Autowired
+  public WebpayplusPaymentServiceImpl(WebpayplusPaymentProperties properties) {
+    this.properties = properties;
+  }
+
+  @Override
+  public PaymentRedirectionDetailsPojo requestNewPaymentPageDetails(SellPojo transaction) throws PaymentServiceException {
+
+
+    String buyOrder = transaction.getBuyOrder().toString();
+    String sessionId = String.valueOf(transaction.hashCode());
+    double amount = transaction.getTotalValue();
+    String returnUrl = properties.getCallbackUrl();
+
+    WebpayPlus.Transaction webpayTransaction = this.createWebpayTransaction();
+
+    try {
+      WebpayPlusTransactionCreateResponse webpayResponse = webpayTransaction.create(
+        buyOrder, sessionId, amount, returnUrl
+      );
+      return PaymentRedirectionDetailsPojo.builder()
+        .url(webpayResponse.getUrl())
+        .token(webpayResponse.getToken())
+        .build();
+    } catch (TransactionCreateException | IOException exc) {
+      logger.error("Exception raised while creating transaction: ", exc);
+      throw new PaymentServiceException("Webpay could not create a new transaction");
+    }
+  }
+
+  @Override
+  public int requestPaymentResult(String transactionToken) throws PaymentServiceException {
+    WebpayPlus.Transaction webpayTransaction = this.createWebpayTransaction();
+    try {
+      return webpayTransaction.commit(transactionToken).getResponseCode();
+    } catch (TransactionCommitException exc) {
+      return 1;
+    } catch (IOException exc) {
+      logger.error("Exception raised while requesting transaction result: ", exc);
+      throw new PaymentServiceException("Webpay failed to confirm the transaction");
+    }
+  }
+
+  @Override
+  public String getPaymentResultPageUrl() {
+    return properties.getBrowserRedirectionUrl();
+  }
+
+  private WebpayPlus.Transaction createWebpayTransaction() {
+    String commerceCode = IntegrationCommerceCodes.WEBPAY_PLUS;
+    String apiKey = IntegrationApiKeys.WEBPAY;
+    IntegrationType integrationType = IntegrationType.TEST;
+    if (properties.isProduction() != null && properties.isProduction()) {

```

Comment:
`(properties.isProduction() != null && properties.isProduction())` this is a non compliant solution. If you check the sonarlink. It suggest to use `Boolean.TRUE.equals(properties.isProduction())` and there is no need to null check

Reply:
Hello @NyorJa, Fixed